### PR TITLE
Droth-303 Bus stops should be added to floating bus stops work list if road link administrative class is changed to 'state'

### PR DESCRIPTION
**Pull request by marquesfm**
- New IF clause for validation of administration Class was changed from Municipality/Private to State
- Created new Bus Stop for Unit Tests;
- Unit tests for all the modifications;
- MassTransitStopService modifications;
- Corrections at Unit Tests;
- Methods for do the updates for the Floating Status and for the Floating Reason;
- Batch for verify if exists modifications in a Road Link Administrative Class.

### CODE REVIEW

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala [hunk 1]
```diff
@@ -396,6 +398,71 @@ object DataFixture {
...
+    println("Processing %d assets not floating".format(assets.length))
+
+    if (assets.length > 0) {
+
+      val roadLinks = vvhClient.fetchVVHRoadlinks(assets.map(_._2).toSet)
+
+      assets.foreach {
+        _ match {
+          case (assetId, linkId, None) =>
+            println("Don't exist Administration Class value at the asset with id %d .".format(assetId))
```
Comment (oappicgi) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: "Mass transit stop with asset-id: %d doesn't have Administration Class value" might be easier to understand

Comment (marquesfm) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: I agree with you Ossi.
I will modify this.

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala [hunk 2]
```diff
@@ -396,6 +398,71 @@ object DataFixture {
...
+    }
+
+    println("\n")
+    println("Complete at time: ")
+    println(DateTime.now())
+    println("\n")
+  }
+
+  private def verifyIsChanged(propertyPublicId: String, propertyId: Long, municipality: Int): Unit = {
+    case class AssetBeforeUpdate(asset: PersistedPointAsset, persistedFloating: Boolean, floatingReason: Option[FloatingReason])
```
Comment (leirasf) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: This class is not used.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/MassTransitStopService.scala [hunk 3]
```diff
@@ -718,3 +721,30 @@ trait MassTransitStopService extends PointAssetOperations {
...
+  }
+
+  def floatingReason(administrationClass: AdministrativeClass, roadLink: VVHRoadlink): String = {
+    if (administrationClass != null && administrationClass != Unknown) {
+      if ((administrationClass == State && (roadLink.administrativeClass == Municipality || roadLink.administrativeClass == Private))
+        || ((administrationClass == Municipality || administrationClass == Private) && (roadLink.administrativeClass == State))) {
+        return "Road link administration class have changed from %d to %d".format(roadLink.administrativeClass.value, administrationClass.value)
+      }
+    }
+    return null
```
Comment (tvenhola) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/MassTransitStopService.scala]: We should avoid null values like plague. Use Option[T] where applicable.

**Comment by tvenhola** [comment]
Okay for me

**Comment by tvenhola** [comment]
Merging

**Pull request merged**